src/mds/CInode.h

```diff
@@ -1118,7 +1118,7 @@ class CInode : public MDSCacheObject, public InodeStoreBase, public Counter<CIno
   // client caps
   client_t loner_cap = -1, want_loner_cap = -1;
 
-  uint32_t snap_update_ref = 0;
+  std::set<mds_rank_t> snap_update_ref_set;
```

Comment:
can we use `std::unordered_set<>` instead for O(1) access ?
though there won't be millions of replicas to cause performance problems using `std::set<>`

Reply:
Yeah, this looks good to me. I will revise this. Thanks @mchangir

Reply:
@chrisphoffman just pointed me at this PR and I noticed this comment. unordered_hash is a hash map and they are quite large, so it is VERY BAD to put those in to MDSCacheObjects.
How much space does an unordered_set take up? We do want to pay some attention to their size given we have hundreds of thousands or millions of them in the cache...

Reply:
Given that this holds the peer mds rank for which snap update has been sent, it can be a 8bit value (char) which can accommodate MAX_MDS.